packages/react-ui-components/src/components/CodeEditor/CodeEditor.tsx

```diff
@@ -1,53 +1,42 @@
-import React, { useCallback, useEffect, useMemo, useState } from 'react';
+import React, { useCallback, useEffect, useMemo } from 'react';
 import Editor, { useMonaco } from '@monaco-editor/react';
-import { DEFAULT_EDITOR_LANGUAGE, DEFAULT_EDITOR_THEME, DEFAULT_PADDING_TOP, FONT_NOT_FOUND_ERROR } from './config';
+import { DEFAULT_EDITOR_LANGUAGE, DEFAULT_EDITOR_THEME, DEFAULT_PADDING_TOP, KELVIN_CODE_EDITOR_THEME } from './config';
 import { ECodeEditorTheme, ICodeEditorProps, OnEditorChangeCallback } from './types';
 import { KvLoader } from '../stencil-generated';
 import { getEditorOptions } from './utils';
+import { useFontsApi } from '../../utils/useFont/useFontsApi';
 
 export const KvCodeEditor = ({
 	code,
 	readOnly = false,
 	loadingComponent = <KvLoader isLoading />,
 	language = DEFAULT_EDITOR_LANGUAGE,
 	theme = DEFAULT_EDITOR_THEME,
-	customTheme,
+	customTheme = KELVIN_CODE_EDITOR_THEME,
 	paddingTop = DEFAULT_PADDING_TOP,
 	onChange
 }: ICodeEditorProps) => {
 	const monaco = useMonaco();
-	const [isFontLoaded, setFontLoaded] = useState(false);
-
 	const options = useMemo(() => getEditorOptions({ readOnly, padding: { top: paddingTop } }), [readOnly, paddingTop]);
+	const { checkFont, loadFont } = useFontsApi(options.fontFamily, options.fontSize);
 
-	const loadFont = useCallback(async () => {
-		try {
-			const { fontSize, fontFamily } = options;
-			await document.fonts.load(`${fontSize}px ${fontFamily}`);
-			setFontLoaded(true);
-		} catch (error) {
-			throw new Error(FONT_NOT_FOUND_ERROR);
-		}
-	}, [options.fontSize, options.fontFamily]);
-
-	const setupEditor = useCallback(() => {
+	const loadCustomTheme = useCallback(() => {
 		if (!monaco || !customTheme) return;
 		monaco.editor.defineTheme(ECodeEditorTheme.Custom, customTheme);
-		monaco.editor.setTheme(ECodeEditorTheme.Custom);
 	}, [monaco, customTheme]);
 
 	const onTextChange: OnEditorChangeCallback = useCallback(value => onChange?.(value), [onChange]);
 
 	useEffect(() => {
 		loadFont();
-		setupEditor();
-	}, [loadFont, setupEditor]);
+		loadCustomTheme();
+	}, [loadFont, loadCustomTheme]);
 
-	if (!isFontLoaded) {
+	if (!checkFont()) {
```

Comment:
Should you also check here for the "monaco" to be available and for the theme to be loaded?